edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/EdgeHubDockerModule.cs

```diff
@@ -36,5 +36,28 @@ public class EdgeHubDockerModule : DockerModule, IEdgeHubModule
             DefaultValueHandling = DefaultValueHandling.Populate)]
         [DefaultValue(RestartPolicy.Always)]
         public override RestartPolicy RestartPolicy { get; }
+
+        public override bool Equals(IModule<DockerConfig> other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            // EdgeHub Equal() is similar to base class except, we ignore the version since we don't ever use it.
+            return string.Equals(this.Name, other.Name) &&
```

Comment:
extract these lines (common fields) as a protected method e.g. "BaseEquals" and can be used in both EdgeHubDockerModule and DockerModule.  Nit: also suggest to rename "IsEnvDictionaryEqual" to "AreEnvironmentVariablesEqual".